src/components/SingleArticle.jsx

```diff
@@ -2,15 +2,36 @@ import { useEffect, useState } from "react"
 import { useParams } from "react-router-dom"
 import { getArticlebyId } from "../utils/api_funcs"
 import Errors from './Errors'
+import { incVotes } from "../utils/api_funcs"
 
 const SingleArticle = () => {
     const [selectedArticle, setSelectedArticle] = useState({})
     const {article_id} = useParams()
     const [isLoading, setIsLoading] = useState(true)
     const [error, setError] = useState(null)
+    const [voteCount, setVoteCount] = useState(0)
+    const [hasVoted, setHasVoted] = useState(false)
+
+    const handleVoteClick = () => {
+        setVoteCount((currVoteCount) => {
+        return currVoteCount+1})
+        setHasVoted(true)
+        incVotes(article_id).catch((err)=> {
+            setVoteCount((currVoteCount) => {
+                return currVoteCount-1
+            });
+            setError({error:{errorMessage: 'something went wrong'}})
+        });
+}
+    const undoVote =() => {
+        setVoteCount((currVoteCount) => {
+                return currVoteCount-1
+            })
+        setHasVoted(false);
+    }
  
```

Comment:
Voting is slightly broken, when I vote and then refresh the page I see a different vote count rendered. I think this only happens if I click the undo vote button at any point. Looks like the undo vote button optimistically renders but doesn't actually change the API data.